go/vt/vtgate/planbuilder/operators/aggregator.go

```diff
@@ -356,7 +356,7 @@ func (a *Aggregator) addIfAggregationColumn(ctx *plancontext.PlanningContext, co
 			if _, srcIsAlsoAggr := a.Source.(*Aggregator); srcIsAlsoAggr {
 				return 0, vterrors.VT12001("aggregation on top of aggregation not supported")
 			}
-			return -1, vterrors.VT13001(fmt.Sprintf("aggregation column on wrong index: want: %d, got: %d", colIdx, offset))
+			return -1, vterrors.VT12001(fmt.Sprintf("complex aggregation expression: %s", sqlparser.String(aggr.Original)))
```

Comment:
I don't love this error. The problem is often not with the individual expression but rather with derived tables and unions and stuff. The issue is definitely not because the expression is complex.

What about:

```go
vterrors.VT12001("failed to plan aggregation on: %s", sqlparser.String(aggr.Original))
```